did_it_run/tests/lib_tests.rs

```diff
@@ -0,0 +1,35 @@
+extern crate did_it_run;
+
+const EXIT_WITH_ARGUMENT_COUNT: &str = "exit $#";
+const EXIT_WITH_STATUS: &str = "exit $0";
+
+#[test]
+fn exits_with_status() {
+    let base_args = vec!["bash", "-c", EXIT_WITH_STATUS];
+    for status_code in vec![-7, -1, 0, 1, 2, 254, 255, 256, 2000] {
+        let status_arg = status_code.to_string();
+        let mut args = base_args.clone();
+        args.push(&status_arg);
+        let status = did_it_run::run_command("/usr/bin/env", args);
+        // When a parent retrieves the exit status of its child, only the
+        // least-significant eight bits are available.
+        let status_code = status_code & 0xFF;
```

Comment:
This would read better as a binary literal.

Reply:
I personally find this a bit more difficult to read because it isn't immediately obvious how many ones are written (I need to squint and count with my cursor). Formally this problem is called [Subitizing](https://en.wikipedia.org/wiki/Subitizing) where people encounter difficulties judging the number of displayed items (e.g. characters) beyond four.

> The accuracy, speed, and confidence with which observers make judgments of the number of items are critically dependent on the number of elements to be enumerated. Judgments made for displays composed of around **one to four items are rapid, accurate and confident**. However, as the number of items to be enumerated increases beyond this amount, judgments are made with decreasing accuracy and confidence. In addition, response times rise in a dramatic fashion, with an **extra 250–350 ms added for each additional item within the display beyond about four**.
https://en.wikipedia.org/wiki/Subitizing

```
let status_code = status_code & 0xFF;
let status_code = status_code & 0b11111111;
```

The comment helps in both cases:
* For `0xFF` it is easy to judge there are exactly two `F` characters. The comment reminds readers that `0xFF`represents eight ones if that isn't immediately obvious.
* For `0b11111111` it is difficult to quickly judge how many ones are written. If the reader chooses to read the command they might trust that eight ones are written. Otherwise, they will need to expend mental effort to count the number of ones to determine the mask.

For bit masks other than eight ones I agree the binary literal is considerably more readable. However, I'd argue `0xFF == 0b11111111` is common knowledge and is therefore the shorter representation more readable.

My preference still leans towards `0xFF` but the most important thing is that others can read the code.

Thoughts?

Reply:
Alright, fair enough. I think in in this case I'm fine with `0xFF` then 👍